x-pack/plugins/ml/public/application/model_management/test_models/test_flyout.tsx

```diff
@@ -79,37 +86,41 @@ export const TestTrainedModelFlyout: FC<Props> = ({ model, onClose }) => {
             </>
           ) : null}
 
-          <EuiTabs
-            size="m"
-            css={{
-              marginTop: `-${mediumPadding}`,
-            }}
-          >
-            <EuiTab
-              isSelected={inputType === INPUT_TYPE.TEXT}
-              onClick={() => setInputType(INPUT_TYPE.TEXT)}
-            >
-              <FormattedMessage
-                id="xpack.ml.trainedModels.testModelsFlyout.textTab"
-                defaultMessage="Test using text"
-              />
-            </EuiTab>
-            <EuiTab
-              isSelected={inputType === INPUT_TYPE.INDEX}
-              onClick={() => setInputType(INPUT_TYPE.INDEX)}
-            >
-              <FormattedMessage
-                id="xpack.ml.trainedModels.testModelsFlyout.indexTab"
-                defaultMessage="Test using existing index"
-              />
-            </EuiTab>
-          </EuiTabs>
+          {onlyShowTab === undefined ? (
+            <>
+              <EuiTabs
+                size="m"
+                css={{
+                  marginTop: `-${mediumPadding}`,
+                }}
+              >
+                <EuiTab
+                  isSelected={inputType === INPUT_TYPE.TEXT}
+                  onClick={() => setInputType(INPUT_TYPE.TEXT)}
+                >
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.testModelsFlyout.textTab"
+                    defaultMessage="Test using text"
+                  />
+                </EuiTab>
+                <EuiTab
+                  isSelected={inputType === INPUT_TYPE.INDEX}
+                  onClick={() => setInputType(INPUT_TYPE.INDEX)}
+                >
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.testModelsFlyout.indexTab"
+                    defaultMessage="Test using existing index"
```

Comment:
Not specific to this PR, but as discussed, would be good to look at restricting this to only show indices with text fields:

<img width="526" alt="image" src="https://github.com/elastic/kibana/assets/7405507/ae74d93c-794c-4adf-8360-f5b3bc7a7234">

Reply:
This affects other types of models, so leave checking for indices with `text` fields for a future bit of work.